src/ui/media/MediaLibrary.as

```diff
@@ -710,7 +716,13 @@ spriteFeaturesFilter.visible = false; // disable features filter for now
 				setTimeout(function():void {
 					MP3Loader.convertToScratchSound(sndName, data, function(s:ScratchSound):void {
 						snd = s;
-						startSoundUpload(s, origName, uploadComplete);
+						if(snd && snd.sampleCount > 0) {
+							startSoundUpload(s, origName, uploadComplete);
+						}
+						else{
+							app.removeLoadProgressBox();
+							DialogBox.notify('Error decoding sound', 'Sorry, Scratch was unable to load the sound '+sndName+'.', Scratch.app.stage);
+						}
 					});
```

Comment:
Would you mind either factoring the callback out into its own named function (could still be local if you want -- doesn't matter to me) or moving the new code into `startSoundUpload`? That way the error check and message would only be in one place.